src/check_result.jl

```diff
@@ -1,3 +1,83 @@
+# For once you have the sensitivity by two methods (e.g  both finite-differencing and  AD)
+# the code here checks it is correct.
+# Goal is to only call `@isapprox` on things that render well
+# Note that this must work well both on Differnetial types and Primal types
+
+"""
+    _check_equal(actual, expected; kwargs...)
+
+`@test`'s  that `actual ≈ expected`, but breaks up data such that human readable results
+are shown on failures.
+All keyword arguments are passed to `isapprox`.
+"""
+function _check_equal(
```

Comment:
I wonder if we should export this, (and rename it)
for people who need to write special tests?
Might make it feel more safe to remove `isapprox` as there is a handy subsitution

Reply:
As I think we're probably committed to maintaining some functionality of this form, I agree that we should export it.